doc/tutorials/space_shooter/app/lib/step4/main.dart

```diff
@@ -71,19 +71,20 @@ class Player extends SpriteAnimationComponent
 
     position = game.size / 2;
 
-    _bulletSpawner = TimerComponent(
+    _bulletSpawner = SpawnComponent(
       period: .2,
-      onTick: () {
+      selfPositioning: true,
+      factory: (index) {
         final bullet = Bullet(
```

Comment:
I guess it can return this directly instead of just defining it and then returning it?